youtube_dl/extractor/teleportal.py

```diff
@@ -0,0 +1,40 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+
+class TeleportalIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?teleportal\.ua(/ua)?/(?P<id>[0-9a-z-/]+)'
+    _TEST = {
+        'url': 'https://teleportal.ua/ua/show/stb/master-cheff/bitva-sezonov/vypusk-3',
+        'md5': '07bd056c45b515fa9cc0202b8403df41',
+        'info_dict': {
+            'id': 'show/stb/master-cheff/bitva-sezonov/vypusk-3',
+            'ext': 'mp4',
+            'title': 'МастерШеф. Битва сезонів 3 випуск: найогидніший випуск сезону!',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'description': r're:^<p>Не пропустіть.*',
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        backend_url = 'https://tp-back.starlight.digital/ua/{}'.format(video_id)
+        metadata = self._download_json(backend_url, video_id)
+        api_metadata = self._download_json('https://vcms-api2.starlight.digital/player-api/{}?referer=https://teleportal.ua/&lang=ua'.format(metadata['hash']), video_id)
```

Comment:
Above you set the url in a var separately. Stick to one or the other: it's probably better to put it in two lines so that it's easy to see which line might have failed. So let's try something like this:
```
        api_hash = metadata['hash']
        api_metadata_url = 'https://vcms-api2.starlight.digital/player-api/{0}?referer=https://teleportal.ua/&lang=ua'.format(api_hash)
        api_metadata = self._download_json(api_metadata_url, video_id)
```
Also, after the first line succeeds, it's safe to use `metadata.get()` later.